### PR TITLE
refactor code of index page

### PR DESCRIPTION
**Pull request by argshook**
This PR adjusts the code of the index page in order for it to be easier
to maintain.

No visual or functional changes were done.

This is done for easier code reuse. For example, `<Section>` and
`<Section.Title>` is used instead of a something like `<div className={seCss["section__title--wide"]}>`

---

commit messages:
- refactor 'why questdb' section
- reuse Section for "Interactive Console" part
- extract QueryScroller to component
- use size="small" for title
- use Section component
- refactor feature tabs
- refactor use cases section

### CODE REVIEW

File: src/components/QueryScroller/index.tsx [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+    if (windowWidth != null && windowWidth < 800) {
+      setTop(M)
+      return
+    }
+
+    setTop(L)
+  }, [windowWidth])
+
+  return (
+    <section
```
Comment (insmac) [inline src/components/QueryScroller/index.tsx:85]: Is it possible to use the `<Section>` component here?

File: src/components/QueryScroller/index.tsx [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+      className={clsx(
+        seCss.section,
+        seCss["section--inner"],
+        seCss["section--center"],
+        seCss["section--showcase"],
+      )}
+    >
+      <h2
```
Comment (insmac) [inline src/components/QueryScroller/index.tsx:93]: Same for `<Section.Title>` ?

**Review by insmac — APPROVED** [review]
Minor comments

File: src/components/QueryScroller/index.tsx (hunk 1 above)
Comment (argshook) [reply src/components/QueryScroller/index.tsx:85]: i started to do tweaks here in this file and saw that it became quite big
so just wanted to leave this for a separate PR.

**Pull request merged**